src/projects.h

```diff
@@ -165,168 +167,317 @@ typedef struct { double u, v; } projUV;
+    int  is_geocent;                /* proj=geocent ... not really a projection at all */
+
+    enum pj_io_units left;          /* Flags for input/output coordinate types */
+    enum pj_io_units right;
+
+
+    /*************************************************************************************
+
+                       C A R T O G R A P H I C       O F F S E T S
+
+    **************************************************************************************/
+
+    double  lam0, phi0;                /* central longitude, latitude */
+    double  x0, y0;                    /* false easting and northing */
+
+
+
+    /*************************************************************************************
+
+                                    S C A L I N G
+
+    **************************************************************************************/
+
+    double  k0;                        /* General scaling factor - e.g. the 0.9996 of UTM */
+    double  to_meter, fr_meter;        /* Plane coordinate scaling. Internal unit [m] */
+    double  vto_meter, vfr_meter;      /* Vertical scaling. Internal unit [m] */
+
+
+
+    /*************************************************************************************
+
+                  D A T U M S   A N D   H E I G H T   S Y S T E M S
+
+    **************************************************************************************/
+
+    int     datum_type;                /* PJD_UNKNOWN/3PARAM/7PARAM/GRIDSHIFT/WGS84 */
+    double  datum_params[7];           /* Parameters for 3PARAM and 7PARAM */
+    struct _pj_gi **gridlist;          /* TODO: Description needed */
+    int     gridlist_count;
+
+    int     has_geoid_vgrids;          /* TODO: Description needed */
+    struct _pj_gi **vgridlist_geoid;   /* TODO: Description needed */
+    int     vgridlist_geoid_count;
+
+    double  from_greenwich;            /* prime meridian offset (in radians) */
+    double  long_wrap_center;          /* 0.0 for -180 to 180, actually in radians*/
+    int     is_long_wrap_set;
+    char    axis[4];                   /* TODO: Description needed */
+
+
+    /* New Datum Shift Grid Catalogs */
+    char   *catalog_name;
+    struct _PJ_GridCatalog *catalog;
+
+    double  datum_date;                 /* TODO: Description needed */
+
+    struct _pj_gi *last_before_grid;    /* TODO: Description needed */
+    PJ_Region     last_before_region;   /* TODO: Description needed */
+    double        last_before_date;     /* TODO: Description needed */
+
+    struct _pj_gi *last_after_grid;     /* TODO: Description needed */
+    PJ_Region     last_after_region;    /* TODO: Description needed */
+    double        last_after_date;      /* TODO: Description needed */
+
+};
+
+
+
+
+
+
+/* Parameter list (a copy of the +proj=... etc. parameters) */
+struct ARG_list {
+    paralist *next;
+    char used;
+    char param[1];    /* This probably should be [0] to be fully standards compliant? */
```

Comment:
No, this is correct; `[0]` would be an empty array. What the code does is allocate the size of this structure + the length of the string. The extra 1 here accounts for the NUL terminator on the string.